src/devices/HuskyLens/DataFrame.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Linq;
+
+namespace Iot.Device.HuskyLens
+{
+    internal class DataFrame
+    {
+        private readonly byte[] _data;
+
+        public DataFrame(ReadOnlySpan<byte> data)
+        {
+            _data = data.ToArray();
+        }
+
+        public Command Command
+        {
+            get
+            {
+                return (Command)_data[4];
+            }
+        }
+
+        public ReadOnlySpan<byte> Data
+        {
+            get
+            {
+                return new ReadOnlySpan<byte>(_data, 5, _data[3]);
+            }
+        }
+
+        public bool Valid()
+        {
+            return
+            _data[0] == 0x55 &&
+            _data[1] == 0xAA &&
+            _data[2] == 0x11 &&
+            _data[3] == _data.Length - 6 &&
+            _data[_data.Length - 1] == CalculateChecksum();
+        }
+
+        /// <inheritdoc/>
+        public override string ToString()
+        {
+            return $"Frame: c={Command}, d={string.Join(" ", Data.ToArray().Select(d => $"{d:X}"))}";
+        }
+
+        private byte CalculateChecksum()
+        {
+            return (byte)(_data.Take(_data.Length - 1).Aggregate(0x00, (a, b) => a + b) & 0xFF);
```

Comment:
I love Linq but is there any other way to avoid using Linq? Reason why is that we're working on making our bindings available for nanoFramework with automatic code transformation and Linq is not supported on this small platform.